### DISCUSSION TITLE
Ability to have docker container use screen when launching.

### DISCUSSION THREAD

**Discussion by JacobT5253** [itzg/docker-minecraft-server#3190] [Ideas]
Ihave a particularly limited use case. but im running this off of my nas from docker desktop. from what I can gather, when i launch the container, i dont have options for modifying how the container is launched. THis is problematic because since the jar file is run at startup, i dont have a good way of accessing the MC terminal though the docker app. it would be nice if i could modify this to launch with the screen command, that way id be able to see the minecraft terminal in another terminal window. Might be possible without a specific update and maybe im not doing it right on my side, but when the docker container was running, i tried to install screen with another terminal and then edited the start.sh file to run the jar file with screen. but it wont launch like that.

**Comment by itzg** [comment]
Docker negates or at least diminishes the need for screen. Have you read this section yet? Do none of these options work for you?

https://docker-minecraft-server.readthedocs.io/en/latest/commands/

**Comment by JacobT5253** [comment]
So due to the limitations of my NAS software I don’t have access to the
terminal on the machine.


I have the docker app. Which doesn’t allow for custom parameters. From what
I can gather, it just launches the docker container with the basic docker
run command.

Since I do not have access to the device terminal, it really limits what I
can do. When the docker container is launched, from within the docker app I
can see the logs for the container, and I have a terminal tab where I can
launch a bash terminal, presumably from within the docker container. I can
also see that there is a Minecraft process running. Regardless, I was just
trying to brainstorm ways I can use my limited setup to have the ability to
view the Minecraft terminal from another terminal window from within the
docker container. Ik using the docker app isn’t ideal but since I don’t
have the CLI option, it sort of sucks. I can get the server up and running
with the docker image, and I can see the logs, I just have no way to access
the Minecraft terminal.

Theoretically I can use rcon, but again since I’m not able to use the
device terminal, it kinda sucks.


My NAS allows for me to access outside of local network via their web
portal. Through this I can view my NAS as a desktop and manage my docker
containers virtually. Which is why I wanted to be able to access the
Minecraft terminal from within the docker container, as it would allow me
to use the NAS desktop connection to connect to the server.

For clients who want to join and play on the server I would port forward
the mc port and link to 3rd party ddns server.

It’s just an idea, maybe not possible to get access to the mc terminal via
the docker app.

There is also the docker image for the Mcroncli, but I can’t seem to get
that one working, and idk if it would work if the mc server is in one
docker container, and the rcon cli is in another.

**Reply by itzg** [reply]
You seem to have a uniquely challenging scenario especially since you don't want to use the existing solutions, such as opening the terminal and using rcon-cli.

Yes, rcon across containers is perfectly fine. It's what allows [the backup image](https://github.com/itzg/docker-mc-backup) to connect and other tools like https://github.com/itzg/docker-rcon-web-admin

**Comment by JacobT5253** [comment]
Ok I’m gonna see if I can use the rcon cli container to access it.
And again if it was as easy as using the device terminal it would already
be done, but because of the systems OS, I don’t get that capability
